src/main/java/io/dockstore/provision/DOSPluginUtil.java

```diff
@@ -36,8 +40,20 @@
      * @return the scheme, host, and path of the targetPath, or <code>Optional.empty()</code>
      */
     Optional<ImmutableTriple<String, String, String>> splitUri(String dosURI) {
+        System.out.println("splitUri: Parsing URI " + dosURI);
         if (Pattern.compile(":\\/\\/(.+)/").matcher(dosURI).find()){
             List<String> split  = Lists.newArrayList(dosURI.split(":\\/\\/|/"));
+
+            // Find out if the path is of the DOS GUID format
+            // dos://dg.4503/6f9ad7df-3751-4056-9340-aa9448525b54
+            System.out.println("splitUri: splitting DOS URI " + dosURI + " Host is " + split.get(HOST));
+            // See if the Host portion starts with 'dg' and ends with a port number
+            List<String> host_split = Lists.newArrayList(split.get(HOST).split("\\.", 2));
+            if (host_split.size() > 1 && host_split.get(0).equals("dg") && NumberUtils.isNumber(host_split.get(1))) {
+                 System.out.println("splitUri: Format is dg.<port> format URI; use " + DG_HOST + " for server");
+                // Construct a URL of the form: https://dataguids.org/ga4gh/dos/v1/dataobjects/dg.4503/6f9ad7df-3751-4056-9340-aa9448525b54
```

Comment:
You're not constructing the URL here, but on line 97

Reply:
Removed comment